File: src/hash/Test1.java
package hash;

import java.util.HashMap;

public class Test1 {
    public static void main(String[] args) {
        String[] participant = {"leo", "kiki", "eden", "leo", "leo"};
        String[] completion = {"eden", "kiki", "leo", "leo"};
        System.out.println("answer : " + solution(participant, completion));
    }


    private static String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap checkMap = new HashMap<String, Integer>();
        for (String value : completion) {
            if (checkMap.get(value) == null) {
                checkMap.put(value, 0);
            } else {
                checkMap.put(value, (Integer) checkMap.get(value) + 1);
            }
        }
        for (String value : participant) {
            if (checkMap.get(value) == null) {
                answer = value;
                break;
            }
            if ((Integer) checkMap.get(value) == 0) {
                checkMap.remove(value);
                continue;
            }
            checkMap.put(value, (Integer) checkMap.get(value) - 1);
        }
        return answer;
    }
}

File: src/hash/Test1.kt
package hash

import java.util.HashMap

fun main() {
    val participants = listOf("leo", "kiki", "eden")
    val completions = listOf("eden", "kiki")
    var answer = ""
    val checkMap = HashMap<String, Int>()
    completions.forEach {
        if (checkMap[it] == null) {
            checkMap[it] = 0
            println("completions null : ${it} / ${checkMap[it]}")
        } else {
            checkMap[it] = checkMap[it] as Int + 1
            println("completions not null : ${it} / ${checkMap[it]}")
        }
    }
    run loop@{
        participants.forEach {
            println("participants : ${it} / ${checkMap[it]}")
            if (checkMap[it] == null) {
                println("participants null : ${it} / ${checkMap[it]}")
                answer = it
                return@loop
            }
            if (checkMap[it] as Int == 0) {
                checkMap.remove(it)
                return@forEach
            }
            println("participants checkMap[it] not 0 : ${it} / ${checkMap[it]}")
            checkMap[it] = checkMap[it] as Int - 1
        }
    }

    println("answer : ${answer}")
}